Python/Product/Debugger/DebugAdapterProcess.cs

```diff
@@ -101,8 +109,22 @@ public sealed class DebugAdapterProcess : ITargetHostProcess {
             _process.ErrorDataReceived += OnErrorDataReceived;
             _process.Start();
             _process.BeginErrorReadLine();
-            if (!_connectedEvent.WaitOne(_debuggerConnectionTimeout)){
-                Debug.WriteLine("Timed out waiting for debuggee to connect.", nameof(DebugAdapterProcess));
+
+            try {
+                if (connection.Wait(_debuggerConnectionTimeout)) {
+                    var socket = connection.Result;
+                    if (socket != null) {
+                        _stream = new NetworkStream(connection.Result, ownsSocket: true);
+                    }
+                } else {
+                    Debug.WriteLine("Timed out waiting for debuggee to connect.", nameof(DebugAdapterProcess));
+                }
+            } catch (AggregateException ex) {
+                Debug.WriteLine("Error waiting for debuggee to connect {0}".FormatInvariant(ex.InnerException ?? ex), nameof(DebugAdapterProcess));
```

Comment:
`_listenerPort` isn't reset to `-1` in case of exception. Is that ok?

Reply:
Yeah, an instance of this class is not reused if it fails.